### PR TITLE
gnrc_tftp: remove module

### PR DESCRIPTION
**Pull request by nmeum**
### Contribution description

Has been deprecated for awhile
(01fc3d8f0bc09fe95cb2b5a9c6bdecb47603f339), has known issues and is poorly maintained.

### Testing procedure

None.

### Issues/PRs references

* #11795
* #11772
* #10927
* …

### CODE REVIEW

**Comment by miri64** [comment]
The deprecation message is quite clear when it is to be removed. So this PR has to wait a bit.

**Comment by nmeum** [comment]
Sure, just wanted to make sure this isn't forgotten.

**Comment by miri64** [comment]
I'm not sure if it makes sense to do this already (not likely, but you might to rebase before April so the commit would change), but please remember to also update the LOSTANDFOUND.md

**Comment by leandrolanzieri** [comment]
@nmeum We can move forward with this now. Please rebase and add this module to `LOSTANDFOUND.md`.

**Comment by nmeum** [comment]
@nmeum We can move forward with this now.  Please rebase and add this module to `LOSTANDFOUND.md`.

Done, though I would like to emphasize that the deprecation note states that this module should be removed (at the latest) **after** the 2020.04 release.

File: LOSTANDFOUND.md [hunk 1]
```diff
@@ -40,6 +40,14 @@ This way, their names are never removed from the RIOT repository.
 
 # Removed Features
 
+### sys/net/gnrc/application_layer/tftp [0e2a62078850e1ecc74db2db4d639cf2d8fb96d3]
+Author:
+- Sören Tempel <tempel@uni-bremen.de>
```
Comment (miri64) [inline LOSTANDFOUND.md]: `git` blame` lists `DipSwitch <dipswitch@ownage4u.nl>` as the main author of that module.

Comment (miri64) [reply LOSTANDFOUND.md]: (unless you want to be the point of contact for questions regarding that module).

Comment (nmeum) [reply LOSTANDFOUND.md]: Ooooppps, no I don't. Fixed, thanks for pointing that out!

**Pull request merged**